### PR TITLE
Add stylelint on default BO theme

### LINKED ISSUE #22885: [BO] Default theme has no specific SCSS rules
#### Is your feature request related to a problem?

Everything folders on the project has Stylelint enabled, while default theme is chaotic!

#### Describe the solution you'd like

Having default theme with stylelint, with at least important rules enabled

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Every themes has stylelint configured, while the default BO theme has not
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #22885.
| How to test?      | You should check if the default theme has no bugs after this PR (sorry :/)
| Possible impacts? | The whole default BO theme

⚠️ Reviewable after a rebase when https://github.com/PrestaShop/PrestaShop/pull/22772 is merged ⚠️ 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22886)

### CODE REVIEW

**Review by atomiix — DISMISSED** [review]

File: admin-dev/themes/default/scss/controllers/_carrier-wizard.scss [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    li {
+      a {
+        color: $main-color;
+      }
+    }
+  }
+  .stepContainer {
+    position: relative;
+    display: block;
+    //overflow: hidden
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/controllers/_carrier-wizard.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/controllers/_customer-thread.scss [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        margin-bottom: 10px;
+      }
+    }
+  }
+}
+.command-danger {
+  background-color: $brand-danger !important;
+}
+.command-success {
+  background-color: $brand-success !important//avatar;
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/controllers/_customer-thread.scss]:
```suggestion
  background-color: $brand-success !important;
```

File: admin-dev/themes/default/scss/controllers/_login.scss [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  .panel-footer {
+    height: inherit;
+    margin: 0 -20px -20px;
+  }
+  .flip-container {
+    height: 420px;
+    margin-top: 115px;
+    transform-style: preserve-3d;
+    @include perspective(1000px);
+    //@include animate(fadeInDown, 0.3s)
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/controllers/_login.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/partials/_commons.scss [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+.fixed-width-lg {
+  width: 160px !important;
+}
+.fixed-width-xl {
+  width: 200px !important;
+}
+.fixed-width-xxl {
+  width: 250px !important;
+}
+//quick fix for inline hidden content on device width - waiting for bootstrap update https://github.com/twbs/bootstrap/issues/8869
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_commons.scss]: @NeOMakinG Twitter Bootstrap has been updated ?

File: admin-dev/themes/default/scss/partials/_commons.scss [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+  font-size: 2em !important;
+}
+.pagination {
+  @include margin(17px, 0, 0, 0);
+}
+.list-detail {
+  dd:not(:last-child) {
+    margin-bottom: 10px;
+  }
+  //colors
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_commons.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/partials/_content.scss [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+    //data-focus
+  }
+}
+.data-focus {
+  &.data-focus-primary {
+    color: #fff;
+    background-color: $brand-primary;
+    @include border-radius(10px);
+  }
+  //orders
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_content.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/partials/_content.scss [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+        padding-left: 8px;
+      }
+      @include rtl () {
+        padding-right: 8px;
+      }
+    }
+    .form-group {
+      margin-bottom: 8px;
+    }
+    //data-focus
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_content.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/partials/_content.scss [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+  }
+  //orders
+}
+#customer_part {
+  .customerCard {
+    &.selected-customer .panel {
+      color: $brand-success;
+      border: solid 2px $brand-success;
+    }
+    //modal
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_content.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/partials/_content.scss [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+      h3 {
+        margin-top: 0;
+      }
+    }
+  }
+  .modal-footer {
+    .alert {
+      text-align: left;
+    }
+    //todo: remove temp fix for bootstrap RC1 -> final
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_content.scss:282]: Bootstrap is in final ?

File: admin-dev/themes/default/scss/partials/_content.scss [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+      }
+    }
+  }
+}
+#header {
+  .panel-footer {
+    height: 40px !important;
+    margin: 15px 0 0 !important;
+  }
+  //help containers
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_content.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/partials/_select2.scss [hunk 11]
```diff
@@ -0,0 +1,694 @@
...
+  border-bottom-right-radius: 0;
+
+  border-bottom-left-radius: 0;
+  -webkit-box-shadow: 0 1px 0 #fff inset;
+  box-shadow: 0 1px 0 #fff inset;
+}
+
+.select2-dropdown-open.select2-drop-above .select2-choice,
+.select2-dropdown-open.select2-drop-above .select2-choices {
+  //border: 1px solid #5897fb;
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_select2.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/partials/_select2.scss [hunk 12]
```diff
@@ -0,0 +1,694 @@
...
+  background-color: #fff;
+  border: 1px solid #aaa;
+}
+
+.select2-locked {
+  padding: 3px 5px !important;
+}
+
+.select2-container-multi.select2-container-active .select2-choices {
+  //border: 1px solid #5897fb;
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_select2.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/partials/_select2.scss [hunk 13]
```diff
@@ -0,0 +1,694 @@
...
+  border-radius: 0 2px 2px 0;
+}
+
+.form-control.select2-container {
+  height: auto !important;
+  padding: 0;
+}
+
+.form-control.select2-container.select2-dropdown-open {
+  //border-color: #5897FB;
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_select2.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/partials/_tree.scss [hunk 14]
```diff
@@ -0,0 +1,82 @@
+.tree-panel-heading-controls {
+  padding: 5px;
+  margin: -20px -20px 20px;
+  font-size: 1.1em;
+  line-height: 2.2em;
+  //text-transform: uppercase
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_tree.scss]:
```suggestion
```

**Review by Progi1984 — COMMENTED** [review]
Some comments

File: admin-dev/themes/default/scss/partials/_commons.scss (hunk 4 above)
Comment (NeOMakinG) [reply admin-dev/themes/default/scss/partials/_commons.scss]: We can't update it due to SemVer

File: admin-dev/themes/default/scss/partials/_content.scss (hunk 9 above)
Comment (NeOMakinG) [reply admin-dev/themes/default/scss/partials/_content.scss:282]: Can't update it due to semver

**Review by Progi1984 — DISMISSED** [review]

File: admin-dev/themes/default/scss/admin-theme.scss [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+@import "vendor/bi-app/bi-app-ltr";
+
+@import "vendor/animate";
+@import "vendor/font-awesome/font-awesome";
+
+//Bootstrap Core : Variables + Mixins + Reset + Print
+@import "vendor/bootstrap-sass/variables";
+@import "vendor/bootstrap-sass/mixins";
+@import "vendor/bootstrap-sass/normalize";
+//@import "vendor/bootstrap-sass/print"
```
Comment (atomiix) [inline admin-dev/themes/default/scss/admin-theme.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/admin-theme.scss [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+  @import "vendor/bootstrap-sass/scaffolding";
+  @import "vendor/bootstrap-sass/type";
+  @import "vendor/bootstrap-sass/code";
+  @import "vendor/bootstrap-sass/grid";
+  @import "vendor/bootstrap-sass/tables";
+  @import "vendor/bootstrap-sass/forms";
+  @import "vendor/bootstrap-sass/buttons";
+  //Components
+  @import "vendor/bootstrap-sass/component-animations";
+  //@import "vendor/bootstrap-sass/glyphicons"
```
Comment (atomiix) [inline admin-dev/themes/default/scss/admin-theme.scss]:
```suggestion
```

File: admin-dev/themes/default/scss/admin-theme.scss [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+  @import "vendor/bootstrap-sass/button-groups";
+  @import "vendor/bootstrap-sass/input-groups";
+  @import "vendor/bootstrap-sass/navs";
+  @import "vendor/bootstrap-sass/navbar";
+  @import "vendor/bootstrap-sass/breadcrumbs";
+  @import "vendor/bootstrap-sass/pagination";
+  @import "vendor/bootstrap-sass/pager";
+  @import "vendor/bootstrap-sass/labels";
+  @import "vendor/bootstrap-sass/badges";
+  //@import "vendor/bootstrap-sass/jumbotron"
```
Comment (atomiix) [inline admin-dev/themes/default/scss/admin-theme.scss]:
```suggestion
```

**Comment by NeOMakinG** [comment]
@atomiix Out of the scope of the PR but I fixed it, if you find something else, please create an issue and list these, it's not a refacto of the default theme, but a swap from sass to scss

**Review by atomiix — DISMISSED** [review]

File: .github/workflows/lint.yml [hunk 18]
```diff
@@ -12,7 +12,14 @@ jobs:
...
 
-      - run: cd ./admin-dev/themes/new-theme && npm install && npm run scss-lint
+      - name: BackOffice Theme `new-theme`
+        run: cd ./admin-dev/themes/new-theme && npm install && npm run scss-lint
+
+      - name: BackOffice Theme `default`
+        run: cd ./admin-dev/themes/default && npm install && npm run scss-lint
+
+      - name: FrontOffice Theme `classic`
+        run: cd themes/classic/_dev && npm install && npm run scss-lint
```
Comment (Progi1984) [inline .github/workflows/lint.yml]:
```suggestion
        run: cd ./themes/classic/_dev && npm install && npm run scss-lint
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

File: admin-dev/themes/default/scss/controllers/_modules.scss [hunk 19]
```diff
@@ -37,18 +37,20 @@
 
 .hook_panel_header {
   @include margin(0, -10px, 0, -10px)
-	@include padding(0, 10px, 10px, 10px)
-	.hook_name {
+    @include padding(0, 10px, 10px, 10px)
+    .hook_name {
+    padding: 0 4px;
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/controllers/_modules.scss:43]:
```suggestion
      padding: 0 4px;
```

File: admin-dev/themes/default/scss/partials/_select2.scss [hunk 20]
```diff
@@ -21,612 +22,626 @@ Version: 3.4.6 Timestamp: Sat Mar 22 22:30:15 EDT 2014
...
 
 .select2-drop-active {
-    //border: 1px solid #5897fb;
-    border-top: none;
+  border-top: none;
 }
 
 .select2-drop.select2-drop-above.select2-drop-active {
-    //border-top: 1px solid #5897fb;
+  //border-top: 1px solid #5897fb;
```
Comment (Progi1984) [inline admin-dev/themes/default/scss/partials/_select2.scss]: To remove

**Review by Progi1984 — COMMENTED** [review]
Some feedback

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @NeOMakinG & @sarahdib

**Pull request merged**